crypto/digestible/derive/test/tests/enums.rs

```diff
@@ -0,0 +1,364 @@
+    assert_eq!(
+        arg.digest32::<MerlinTranscript>(b"B_test"),
+        [
+            50, 246, 247, 147, 99, 239, 185, 52, 199, 224, 32, 139, 105, 56, 188, 103, 146, 175,
+            74, 71, 151, 42, 18, 123, 38, 163, 180, 85, 22, 128, 126, 177
+        ]
+    );
+}
+
+// Test that a C instance of ExampleEnum is being mapped to AST and hashed as
+// expected
+#[test]
+fn example_enum_c() {
+    let arg = ExampleEnum::C;
+    let expected_ast = ASTNode::from(ASTVariant {
+        context: b"C_test",
+        name: b"ExampleEnum".to_vec(),
+        which: 2,
+        value: Some(Box::new(ASTNode::from(ASTNone { context: b"C" }))),
+    });
+    digestible_test_case_ast("C_test", &arg, expected_ast);
+    assert_eq!(
+        arg.digest32::<MerlinTranscript>(b"C_test"),
+        [
+            36, 116, 29, 101, 185, 20, 3, 39, 50, 65, 57, 25, 15, 236, 167, 119, 53, 69, 6, 19,
+            134, 181, 97, 14, 175, 109, 81, 67, 31, 245, 205, 237
+        ]
+    );
+}
+
+// Test that an A instance of TransEnum is being mapped to AST and hashed as
+// expected
+#[test]
+fn trans_enum_a() {
+    let arg = TransEnum::A(3);
+    let expected_ast = ASTNode::from(ASTPrimitive {
+        context: b"A_test",
+        type_name: b"uint",
+        data: vec![3u8, 0u8],
+    });
+    digestible_test_case_ast("A_test", &arg, expected_ast);
+    assert_eq!(
+        arg.digest32::<MerlinTranscript>(b"A_test"),
+        [
+            196, 66, 116, 1, 87, 75, 116, 241, 250, 78, 20, 22, 208, 227, 89, 118, 121, 77, 109,
+            255, 15, 184, 217, 249, 111, 181, 66, 141, 37, 23, 204, 243
+        ]
+    );
+}
+
+// Test that a B instance of ExampleEnum is being mapped to AST and hashed as
+// expected
+#[test]
+fn trans_enum_b() {
+    let arg = TransEnum::B(ExampleStruct { a: 0, b: 1, c: 2 });
+    let expected_ast = ASTNode::from(ASTAggregate {
+        context: b"B_test",
+        name: b"ExampleStruct".to_vec(),
+        elems: vec![
+            ASTNode::from(ASTPrimitive {
+                context: b"a",
+                type_name: b"uint",
+                data: vec![0u8, 0u8],
+            }),
+            ASTNode::from(ASTPrimitive {
+                context: b"b",
+                type_name: b"uint",
+                data: vec![1u8, 0u8],
+            }),
+            ASTNode::from(ASTPrimitive {
+                context: b"c",
+                type_name: b"uint",
+                data: vec![2u8, 0u8],
+            }),
+        ],
+        is_completed: true,
+    });
+    digestible_test_case_ast("B_test", &arg, expected_ast);
+    assert_eq!(
+        arg.digest32::<MerlinTranscript>(b"B_test"),
+        [
+            101, 107, 59, 55, 249, 37, 221, 203, 13, 36, 193, 85, 104, 62, 192, 193, 243, 144, 224,
+            171, 85, 200, 48, 29, 36, 71, 187, 89, 102, 228, 109, 87
+        ]
+    );
+}
+
+// Test that a C instance of ExampleEnum is being mapped to AST and hashed as
+// expected
+#[test]
+fn trans_enum_c() {
```

Comment:
If I add an empty `E` variant to the enum, this passes
```rust
    let arg = TransEnum::C;
    let expected_ast = ASTNode::from(ASTNone { context: b"C_test" });
    digestible_test_case_ast("C_test", &arg, expected_ast);
    assert_eq!(
        arg.digest32::<MerlinTranscript>(b"C_test"),
        TransEnum::E.digest32::<MerlinTranscript>(b"C_test")
    );
```
I'm not quite clear on when/where the context changes.
Initial thought was that `C` should be part of the digest, but then that would go against the transparent.  I'm not clear on the use for digesting these types, but perhaps there needs to be a warning or check against having more than one `Unit` variant in the same Enum.
Thinking more I'm also getting the impression that multiple variants with the same type or size of unnamed fields would collide.

Reply:
I'm fine with removing support for the unit variant in `digestible(transparent)`.

It seemed like a good idea when I was making this PR, but I don't think it's actually needed in the PR that depends on this. Rust's protobuf crate uses `Option<Enum>` to represent `oneof` fields.

It may be that the support for unit variants is YAGNI

Reply:
Okay we have now removed unit variants